tests/test_regression.py

```diff
@@ -4,40 +4,48 @@
 from pathlib import Path
 from typing import Any
 
+import pytest
+
 from autocorpus.configs.default_config import DefaultConfig
 
 
-def test_autocorpus(data_path: Path):
-    """A regression test for the main Auto-CORPus class on the Auto-CORPus paper.
+@pytest.mark.parametrize(
+    "input_file, config",
+    [
+        ("PMC/Pre-Oct-2024/PMC8885717.html", DefaultConfig.LEGACY_PMC.load_config()),
+        ("PMC/Current/PMC8885717.html", DefaultConfig.PMC.load_config()),
+    ],
+)
+def test_autocorpus(data_path: Path, input_file: Path, config: dict[str, Any]) -> None:
+    """A regression test for the main autoCORPus class, using the each PMC config on the AutoCORPus Paper.
 
     The test data output is built by running Auto-CORPus from the root of the repo with the
     following arguments:
-
     ```
-    auto-corpus -b LEGACY_PMC -t tests/data/PMC/Pre-Oct-2024/ -f tests/data/PMC/Pre-Oct-2024/PMC8885717.html
+    auto-corpus -b "LEGACY_PMC" -t "tests/data/PMC/Pre-Oct-2024/" -f "tests/data/PMC/Pre-Oct-2024/PMC8885717.html"
     ```
     """
     from autocorpus.autocorpus import Autocorpus
 
-    pmc_example_path = data_path / "PMC" / "Pre-Oct-2024"
+    pmc_example_path = data_path / input_file
     with open(
-        pmc_example_path / "PMC8885717_abbreviations.json",
+        str(pmc_example_path).replace(".html", "_abbreviations.json"), encoding="utf-8"
```

Comment:
It's cleaner to use `with_suffix` and `path.open`
```suggestion
       pmc_example_path.with_suffix("_abbreviations.json").open(encoding="utf-8")
```

Reply:
Sorry, just checked and this doesn't quite work. It'll need to be:
```suggestion
        pmc_example_path.parent / f"{pmc_example_path.stem}_abbreviations.json", encoding="utf-8"
```

And then the `.open` is a bit too messy to include

Reply:
I liked the "with_suffix" solution, if it worked ofc, but isn't the second solution just doing the same thing? string conversion (by being inside an f string) and concatenation? Seems we're just swapping a replace with a concatenate at this point

Reply:
The difference is we're not casting the path to a str and then opening the file. It's very minor though, and not super important because we are only changing the end of the filename and not the path separators. So I'll let it go